library/src/main/java/com/liulishuo/filedownloader/connection/FileDownloadUrlConnection.java

```diff
@@ -46,6 +46,9 @@ public FileDownloadUrlConnection(URL url, Configuration configuration) throws IO
         } else {
             mConnection = url.openConnection();
         }
+        if (mConnection instanceof HttpURLConnection) {
```

Comment:
why close?

Reply:
HttpURLConnection will handle redirect automatically, in this way, FileDownloader cannot get redirect url, then use original url to start fetching but get a 302 response.  There is two problems, if the file size is fixed, there is duplicated connection; if the file size is dynamic(different source), the file size is wrong and cannot download successful, like #1295.